app/webpacker/controllers/csv_select_controller.js

```diff
@@ -0,0 +1,21 @@
+import { Controller } from "stimulus";
+
+export default class extends Controller {
+  static targets = ["reportType", "checkbox", "label"]
+
+  handleSelectChange() {
+    this.reportTypeTarget.value == "csv" ? this.disableField() : this.enableField()
```

Comment:
What do you think about making this more generic? If you supplied the value `"csv"` to the controller then it could work with any input field and disable any other form elements.

A generic controller would need to iterate through all targets and apply an action depending on their type. Checkboxes get checked, labels get a class, all inputs and buttons get disabled. This can be extended if needed.

You don't need to do this here now. It's just an idea to build more generic functionality which can be re-used instead of creating new controllers all the time.